src/content/en/fundamentals/primers/shadowdom/index.markdown

```diff
@@ -689,21 +689,108 @@ If the shadow tree is open, calling `event.composedPath()` will return an array
+  tabs.dispatchEvent(new Event('tab-select', {bubbles: true, composed: true}));
+}
+{% endhighlight %}
 
 If `composed: false` (default), consumers won't be able to listen for the event outside of your shadow root.
 
-    <fancy-tabs></fancy-tabs>
-    <script>
-      const tabs = document.querySelector('fancy-tabs');
-      tabs.addEventListener('tab-select', e => {
-        // won't fire if `tab-select` wasn't created with `composed: true`.
-      });
-    </script>
+{% highlight html %}
+<fancy-tabs></fancy-tabs>
+<script>
+  const tabs = document.querySelector('fancy-tabs');
+  tabs.addEventListener('tab-select', e => {
+    // won't fire if `tab-select` wasn't created with `composed: true`.
+  });
+</script>
+{% endhighlight %}
+
+### Handling focus {#focus}
+
+If you recall from [shadow DOM's event model](#events), events that are fired inside shadow DOM are adjusted to look like they come from the hosting element. For example, let's say you click an `<input>` inside a shadow root:
+
+{% highlight html %}
+<x-focus>
+  #shadow-root
+    <input type="text" placeholder="Input inside shadow dom">
+{% endhighlight %}
+
+ The `focus` event will look like it came from `<x-focus>`, not the `<input>`. Similarly, `document.activeElement` will be `<x-focus>`. If the shadow root was created with `mode:'open'` (see [closed mode](#closed)), you'll also be able access the internal node that gained focus:
+
+    document.activeElement.shadowRoot.activeElement // only works with open mode.
+
+Another option is `delegatesFocus: true`, which expands the focus behavior of element's within a shadow tree:
+
+- If you click on a node inside shadow DOM and the node is not a focusable area, the first focusable area becomes focused.
+- When a node inside shadow DOM gains focus, `:focus` applies to the host in addition to the focused element.
+
+**Example** - how `delegatesFocus: true` changes focus behavior
+
+{% highlight html %}
+<style>
+  :focus {
+    outline: 2px solid red;
+  }
+</style>
+
+<x-focus></x-focus>
+
+<script>
+customElements.define('x-focus', class extends HTMLElement {
+  constructor() {
+    super(); // always call super() first in the ctor.
+
+    const root = this.attachShadow({mode: 'open', delegatesFocus: true});
+    root.innerHTML = `
+      <style>
+        :host {
+          display: flex;
+          border: 1px dotted black;
+          padding: 16px;
+        }
+        :focus {
+          outline: 2px solid blue;
+        }
+      </style>
+      <div>Clickable Shadow DOM text</div>
+      <input type="text" placeholder="Input inside shadow dom">`;
+
+    // Know the focused element inside shadow DOM:
+    this.addEventListener('focus', function(e) {
+      console.log('Active element (inside shadow dom):',
+                  this.shadowRoot.activeElement);
+    });
+  }
+});
+</script>
+{% endhighlight %}
+
+**Result**
+
+<img src="images/delegateFocusTrue.png" title="delegatesFocus: true behavior">
```

Comment:
Yea, I want to illustrate the how the styles are affected inside SD and outside of it.